rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -189,25 +191,35 @@ def put(self):
          deployment.
         :param target_deployment: the deployment that the source deployment
          depends on.
+        :param external_source: metadata, in JSON format, of the source
+        deployment (deployment name, tenant name, and the manager host(s)),
+        in case it resides on an external manager. None otherwise
+        :param external_target: metadata, in JSON format, of the target
+        deployment (deployment name, tenant name, and the manager host(s)),
+        in case it resides on an external manager. None otherwise
         :return: an InterDeploymentDependency object containing the information
          of the dependency.
         """
         sm = get_storage_manager()
+
         params = self._get_put_dependency_params(sm)
         now = utils.get_formatted_timestamp()
 
-        # assert no cyclic dependencies are created
-        dep_greph = rest_utils.RecursiveDeploymentDependencies(sm)
-        source_id = str(params[SOURCE_DEPLOYMENT].id)
-        target_id = str(params[TARGET_DEPLOYMENT].id)
-        dep_greph.create_dependencies_graph()
-        dep_greph.assert_no_cyclic_dependencies(source_id, target_id)
+        if (EXTERNAL_SOURCE not in params) and (EXTERNAL_TARGET not in params):
+            # assert no cyclic dependencies are created
+            dep_greph = rest_utils.RecursiveDeploymentDependencies(sm)
+            source_id = str(params[SOURCE_DEPLOYMENT].id)
+            target_id = str(params[TARGET_DEPLOYMENT].id)
+            dep_greph.create_dependencies_graph()
+            dep_greph.assert_no_cyclic_dependencies(source_id, target_id)
 
         deployment_dependency = models.InterDeploymentDependencies(
             id=str(uuid.uuid4()),
             dependency_creator=params[DEPENDENCY_CREATOR],
             source_deployment=params[SOURCE_DEPLOYMENT],
-            target_deployment=params[TARGET_DEPLOYMENT],
+            target_deployment=params.get(TARGET_DEPLOYMENT, None),
```

Comment:
The default value of `dict.get(something)` is `None` so you don't have to add it.

Reply:
1